python/python/bystro/proteomics/fragpipe_tandem_mass_tag.py

```diff
@@ -0,0 +1,53 @@
+"""Load and prep fragpipe tandem mass Tag datasets."""
+
+from dataclasses import dataclass
+
+import pandas as pd
+
+ABUNDANCE_COLS = ["Index", "NumberPSM", "ProteinID", "MaxPepProb", "ReferenceIntensity"]
+ANNOTATION_COLS = ["plex", "channel", "sample", "sample_name", "condition", "replicate"]
+
+
+@dataclass(frozen=True)
+class TandemMassTagDataset:
+    """Represent a Fragpipe Tandem Mass Tag dataset."""
+
+    abundance_df: pd.DataFrame
+    annotation_df: pd.DataFrame
+
+
+def _check_df_cols(df: pd.DataFrame, expected_cols: list[str]) -> None:
+    actual_cols = df.columns
+    if not all(x == y for x, y in zip(expected_cols, actual_cols, strict=False)):
+        err_msg = (
+            f"expected dataframe to begin with cols: {expected_cols}, got cols: {actual_cols} instead."
```

Comment:
General question - is there an overlapping core set of columns between outputs from different fragpipe pipelines or will we have to be very specific about what is allowed as input?

Reply:
The annotation files look to be conserved between pipelines, but the abundance filenames look to be pretty sui generis

Reply:
Is it possible we'll be able to capture at least most/many TMT pipeline outputs with these headers? May be a question we send out to our contacts.